src/sentry/incidents/logic.py

```diff
@@ -893,7 +973,31 @@ def update_alert_rule(
             updated_fields["user_id"] = alert_rule.user_id
             updated_fields["team_id"] = alert_rule.team_id
 
+        if detection_type == AlertRuleDetectionType.DYNAMIC:
+            if not features.has("organizations:anomaly-detection-alerts", alert_rule.organization):
+                raise ResourceDoesNotExist(
+                    "Your organization does not have access to this feature."
+                )
+            if should_update_seer(
+                alert_rule.detection_type,
+                updated_fields.get("detection_type"),
+                query,
+                aggregate,
+            ):
+                update_dynamic_alert_data(
```

Comment:
Does this need to be a separate helper?
the method above seems fairly straight forwards and i don't know if we gain much from componentizing this?

Reply:
`create_alert_rule` and `update_alert_rule` are just SO long